packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -2103,22 +2093,164 @@ describe('Checkout', () => {
                     }
                 });
 
-                const payload = {
-                    action: 'submit'
-                };
+                // Act & Assert
+                expect(wrapper.vm.onFormSubmit).toBeDefined();
+            });
 
-                // Act
-                await wrapper.vm.onFormSubmit();
+            describe('when invoked', () => {
+                it('should make a call to `trackFormInteraction` so we can track the action type `submit`', async () => {
+                    // Arrange
+                    isFormValidSpy.mockReturnValue(true);
+                    const trackFormInteractionSpy = jest.spyOn(VueCheckout.methods, 'trackFormInteraction');
+                    const wrapper = mount(VueCheckout, {
+                        store: createStore(),
+                        i18n,
+                        localVue,
+                        propsData,
+                        mocks: {
+                            $v,
+                            $logger
+                        }
+                    });
 
-                // Assert
-                expect(trackFormInteractionSpy).toHaveBeenCalledWith(payload);
+                    // Act
+                    await wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(trackFormInteractionSpy).toHaveBeenCalledWith({
+                        action: 'submit'
+                    });
+                });
+
+                describe('AND the call to `isFormValid` falsey', () => {
```

Comment:
And here its called when falsey?

Reply:
This one is correct. It's fired when false.